BSP/Src/BSP_UART.c

```diff
@@ -0,0 +1,85 @@
+#include "BSP_UART.h"
+
+static UART_HandleTypeDef for_interrupts;
+
+//TODO This function needs to be reviewed, and queue creation will likely happen here
+UART_Init_Status BSP_UART_Init(USART_HandleTypeDef device) {
+    GPIO_InitTypeDef gpio_struct;
+    UART_InitTypeDef uart_struct;
+    for_interrupts.Instance = USART3;
+
+    BSP_Status returnInfo;
+
+    // Init clocks to port b, c, and usart3
+
+    // check if they're disabled
+    // if so, enable
+    if (__HAL_RCC_GPIOB_IS_CLK_DISABLED()) {
+        __HAL_RCC_GPIOB_CLK_ENABLE();
+    }
+    if (__HAL_RCC_GPIOC_IS_CLK_DISABLED()) {
+        __HAL_RCC_GPIOC_CLK_ENABLE();
+    }
+    if (__HAL_RCC_USART3_IS_CLK_DISABLED()) {
+        __HAL_RCC_USART3_CLK_ENABLE();
+    }
+
+    // config pins
+    gpio_struct.Pin = GPIO_PIN_10;
+    gpio_struct.Alternate = GPIO_AF7_USART3;
+    gpio_struct.Mode = MODE_AF;
+    gpio_struct.Pull = GPIO_PULLUP;
+    gpio_struct.Speed = GPIO_SPEED_FREQ_LOW; // "GPIO Output Maximum Frequency"
+    HAL_GPIO_Init(GPIOB, &gpio_struct);
+    gpio_struct.Pin = GPIO_PIN_5;
+    HAL_GPIO_Init(GPIOC, &gpio_struct);
+
+    // uart config
+    uart_struct.BaudRate = 115200;
+    uart_struct.HwFlowCtl = UART_HWCONTROL_NONE;
+    uart_struct.Mode = UART_MODE_TX_RX;
+    uart_struct.Parity = UART_PARITY_NONE;
+    uart_struct.WordLength = UART_WORDLENGTH_8B;
+    uart_struct.StopBits = UART_STOPBITS_1;
+    returnInfo = CONVERT_RETURN(HAL_UART_Init(&uart_struct));
+
+    __HAL_USART_ENABLE_IT(&for_interrupts, USART_IT_RXNE); //rx
+    __HAL_USART_ENABLE_IT(&for_interrupts, USART_IT_TC); //tx
```

Comment:
i dont think we need these function calls as the HAL_NVIC function calls should do it for us